examples/spawn-and-move/manifests/deployments/KATANA.json

```diff
@@ -0,0 +1 @@
ame\": \"dojo::components::upgradeable::IUpgradeable\"\n  },\n  {\n    \"type\": \"interface\",\n    \"name\": \"dojo::components::upgradeable::IUpgradeable\",\n    \"items\": [\n      {\n        \"type\": \"function\",\n        \"name\": \"upgrade\",\n        \"inputs\": [\n          {\n            \"name\": \"new_class_hash\",\n            \"type\": \"core::starknet::class_hash::ClassHash\"\n          }\n        ],\n        \"outputs\": [],\n        \"state_mutability\": \"external\"\n      }\n    ]\n  },\n  {\n    \"type\": \"event\",\n    \"name\": \"dojo::components::upgradeable::upgradeable::Upgraded\",\n    \"kind\": \"struct\",\n    \"members\": [\n      {\n        \"name\": \"class_hash\",\n        \"type\": \"core::starknet::class_hash::ClassHash\",\n        \"kind\": \"data\"\n      }\n    ]\n  },\n  {\n    \"type\": \"event\",\n    \"name\": \"dojo::components::upgradeable::upgradeable::Event\",\n    \"kind\": \"enum\",\n    \"variants\": [\n      {\n        \"name\": \"Upgraded\",\n        \"type\": \"dojo::components::upgradeable::upgradeable::Upgraded\",\n        \"kind\": \"nested\"\n      }\n    ]\n  },\n  {\n    \"type\": \"event\",\n    \"name\": \"dojo_examples::actions::actions::Moved\",\n    \"kind\": \"struct\",\n    \"members\": [\n      {\n        \"name\": \"player\",\n        \"type\": \"core::starknet::contract_address::ContractAddress\",\n        \"kind\": \"key\"\n      },\n      {\n        \"name\": \"direction\",\n        \"type\": \"dojo_examples::models::Direction\",\n        \"kind\": \"data\"\n      }\n    ]\n  },\n  {\n    \"type\": \"event\",\n    \"name\": \"dojo_examples::actions::actions::Event\",\n    \"kind\": \"enum\",\n    \"variants\": [\n      {\n        \"name\": \"UpgradeableEvent\",\n        \"type\": \"dojo::components::upgradeable::upgradeable::Event\",\n        \"kind\": \"nested\"\n      },\n      {\n        \"name\": \"Moved\",\n        \"type\": \"dojo_examples::actions::actions::Moved\",\n        \"kind\": \"nested\"\n      }\n    ]\n  }\n]","reads":["Moves","Position"],"writes":[],"computed":[],"name":"dojo_examples::actions::actions"}],"models":[{"kind":"DojoModel","members":[{"name":"player","type":"ContractAddress","key":true},{"name":"direction","type":"Direction","key":false}],"class_hash":"0x52659850f9939482810d9f6b468b91dc99e0b7fa42c2016cf12833ec06ce911","abi":null,"name":"dojo_examples::actions::actions::moved"},{"kind":"DojoModel","members":[{"name":"player","type":"ContractAddress","key":true},{"name":"remaining","type":"u8","key":false},{"name":"last_direction","type":"Direction","key":false}],"class_hash":"0x511fbd833938f5c4b743eea1e67605a125d7ff60e8a09e8dc227ad2fb59ca54","abi":null,"name":"dojo_examples::models::moves"},{"kind":"DojoModel","members":[{"name":"player","type":"ContractAddress","key":true},{"name":"vec","type":"Vec2","key":false}],"class_hash":"0xb33ae053213ccb2a57967ffc4411901f3efab24781ca867adcd0b90f2fece5","abi":null,"name":"dojo_examples::models::position"}]}
```

Comment:
i think we should pretty print this

Reply:
I think I've tried and I removed it as I had all those `\n`. Will check it again, nice catch mate.

Reply:
Fixed. The problem was that I was trying to handle the ABI as a string. But to have correct formatting I used the `AbiEntry` type. Which is more appropriate to handle the ABI though.